### DISCUSSION TITLE
system.text.json support for polymorphic serialization.

### DISCUSSION THREAD

**Discussion by qsdfplkj** [dotnet/runtime#40269] [General] [answered]
There has been questions before about polymorphic (de)serialization of json. There is an argument to not support this because it opens a security vulnerability where completely unrelated classes could be instantiated and allow execute of arbitrary code. But why not allow for custom interfaces or types to be used. Just `Deserialize<IMyInterface>` shouldn't be causing problems?

**Comment by layomia** [comment, marked as answer]
Please follow the discussions in https://github.com/dotnet/runtime/issues/29937 and https://github.com/dotnet/runtime/issues/30083 for updates on polymorphic support in System.Text.Json.

**Reply by qsdfplkj** [reply]
I added to the discussion: https://github.com/dotnet/runtime/issues/29937#issuecomment-670581015
